### PR TITLE
7536/improve source pane padding

### LINKED ISSUE #7536: Wrong vertical padding in Sources pane
It might be a regression introduced in the PR for #6804.

The `padding: 4px 0;` we wanted to have on the Sources and Outline panes should be set an element that is either the scrolling context or inside the scrolling context. Currently it's set on a parent, so when a scrollbar appears it does not touch the edges (neither does the content):

screen shot

### PR DESCRIPTION
**Pull request by Garee**
Specify a 4px top/bottom padding for the sources list tree element and remove the top margin from its first child. This fixes the gap between the list scrollbar and the top of the container when it overflows.

Remove the margin from the outline filter and give it the same 4px top padding to maintain consistency across tabs.

Updated the expected test results as the expected styles have changed.

Fixes #7536.

### CODE REVIEW

**Comment by Garee** [comment]
@nchevobbe - This is a continuation of [this pull request](https://github.com/devtools-html/debugger.html/pull/7582).

File: src/components/PrimaryPanes/Sources.css [hunk 1]
```diff
@@ -62,6 +62,7 @@
   display: flex;
   flex: 1;
   height: 100%;
+  padding-bottom: 4px;
```
Comment (nchevobbe) [inline src/components/PrimaryPanes/Sources.css]: i'm not sure why we have this one in addition to the bottom padding we have on the tree as well.
Here it looks like we have a 4px padding at the top, then a 8px one at the bottom, which is a bit odd.

Comment (Garee) [reply src/components/PrimaryPanes/Sources.css]: I removed this padding. Thanks.

**Comment by Garee** [comment]
> Since we have padding on the tree, when there are selected items in it, they don't stick to the border of the panel, looking a bit off:

@nchevobbe What should I change it to? I thought this was correct based on the comments by @fvsch in #7536.

**Comment by fvsch** [comment]
> Since we have padding on the tree, when there are selected items in it, they don't stick to the border of the panel, looking a bit off.

That's an expected result. Not ideal, but the padding has other benefits, so for now it's a trade-off we want. (Can't promise that it'll never ever change in a design review though ^^)

File: src/components/PrimaryPanes/Sources.css [hunk 2]
```diff
@@ -74,8 +74,9 @@
   display: flex;
 }
 
-.sources-list .managed-tree .tree-node:first-child {
-  margin-top: 4px;
+.sources-list .managed-tree .tree {
+  overflow: visible;
```
Comment (nchevobbe) [inline src/components/PrimaryPanes/Sources.css]: do we need this? It seems to work fine without it.

Comment (Garee) [reply src/components/PrimaryPanes/Sources.css]: Nope! I've removed it. Thanks.

**Comment by nchevobbe** [comment]
Thanks @Garee !

**Pull request merged**